assets/in

```diff
@@ -19,17 +17,14 @@ fi
 # parse incoming config data
 payload=`cat`
 bucket=$(echo "$payload" | jq -r '.source.bucket')
+options=$(echo "$payload" | jq -r '.source.options // [] | join(" ")')
 
-
-echo "Downloading from S3..."
-
-# http://docs.aws.amazon.com/cli/latest/userguide/cli-chap-getting-started.html#cli-environment
+# export for `aws` cli
 export AWS_ACCESS_KEY_ID=$(echo "$payload" | jq -r '.source.access_key_id')
 export AWS_SECRET_ACCESS_KEY=$(echo "$payload" | jq -r '.source.secret_access_key')
 
-aws s3 sync "s3://$bucket" $dest
-
+echo "Downloading from S3..."
+eval aws s3 sync "s3://$bucket" $dest $options
```

Comment:
Think of a shell command as an array, where the elements are delimited by whitespace. Without the `eval`, everything in `$options` is evaluated as a single argument, whereas `eval` does the string interpolation (i.e. expanding the `$options`) before it does the splitting of arguments. Example (`$#` in a shell script is the number of arguments):

```
$ echo "echo \$#" > test.sh
$ chmod +x test.sh
$ ./test.sh
0
$ OPTS="foo bar"
$ ./test.sh $OPTS
1
$ eval ./test.sh $OPTS
2
```

Does that make sense?